osu.Game/Database/ArchiveModelManager.cs

```diff
@@ -546,6 +549,11 @@ private List<TFileModel> createFileInfos(ArchiveReader reader, FileStore files)
         /// </summary>
         protected virtual string ImportFromStablePath => null;
 
+        /// <summary>
+        /// Does stable import look for directories rather than files
+        /// </summary>
+        protected abstract bool StableDirectoryBased { get; }
```

Comment:
We're likely going to need custom logic in `BeatmapManager` anyway to do recursive lookups. A better way of doing this is to have a method `IEnumerable<string> GetStableImportPaths()`, which does your logic on a per-manager basis.